### PR TITLE
Update directives parser to include missing documents

### LINKED ISSUE #238: Fails to parse ""ISO/IEC DIR 2 ISO"
```ruby
$ Pubid::Iso::Identifier.parse "ISO/IEC DIR 2 ISO"
Failed to match sequence (stage:'Fpr'? 'WD/'? (type:GUIDE_PREFIX SPACE)? (stage:STAGE SPACE)? (stage:TYPED_STAGE SPACE)? (ORIGINATOR (SPACE / '/'))? (TC_DOCUMENT_BODY / STD_DOCUMENT_BODY / DIR_DOCUMENT_BODY (' + ' dir_joint_document:(ORIGINATOR SPACE DIR_DOCUMENT_BODY))?)) at line 1 char 9.
cause: Failed to match sequence (stage:'Fpr'? 'WD/'? (type:GUIDE_PREFIX SPACE)? (stage:STAGE SPACE)? (stage:TYPED_STAGE SPACE)? (ORIGINATOR (SPACE / '/'))? (TC_DOCUMENT_BODY / STD_DOCUMENT_BODY / DIR_DOCUMENT_BODY (' + ' dir_joint_document:(ORIGINATOR SPACE DIR_DOCUMENT_BODY))?)) at line 1 char 9.
`- Expected one of [TC_DOCUMENT_BODY, STD_DOCUMENT_BODY, DIR_DOCUMENT_BODY (' + ' dir_joint_document:(ORIGINATOR SPACE DIR_DOCUMENT_BODY))?] at line 1 char 9.
   |- Failed to match sequence ((tctype:TCTYPE '/'?){0, } SPACE tcnumber:DIGITS ('/' ((sctype:SCTYPE SPACE scnumber:DIGITS '/')? wgtype:WGTYPE SPACE wgnumber:DIGITS / sctype:SCTYPE (SPACE / '/' wgtype:WGTYPE SPACE) scnumber:DIGITS))? SPACE 'N' SPACE? number:DIGITS) at line 1 char 9.
   |  `- Expected " ", but got "D" at line 1 char 9.
   |- Failed to match sequence ((TYPE / stage:STAGE iteration:DIGITS?)? SPACE? ((stage:STAGE / stage:TYPED_STAGE / TYPE) SPACE)? number:DIGITS ('|' joint_document:(publisher:'IDF' SPACE number:DIGITS))? PART? ITERATION? (SPACE? (':' / DASH) YEAR)? SUPPLEMENT? EXTRACT? ADDENDUM? EDITION? LANGUAGE?) at line 1 char 14.
   |  `- Extra input after last repetition at line 1 char 14.
   |     `- Failed to match sequence ('(' language:(([a-z]{1, } ','? / ('E' / 'F' / 'A' / 'R') '/'?){0, }) ')') at line 1 char 14.
   |        `- Expected "(", but got " " at line 1 char 14.
   `- Failed to match sequence (DIR_DOCUMENT_BODY (' + ' dir_joint_document:(ORIGINATOR SPACE DIR_DOCUMENT_BODY))?) at line 1 char 14.
      `- Extra input after last repetition at line 1 char 14.
         `- Failed to match sequence (' + ' dir_joint_document:(ORIGINATOR SPACE DIR_DOCUMENT_BODY)) at line 1 char 14.
            `- Expected " + ", but got " IS" at line 1 char 14.
```
Blocks this https://github.com/relaton/relaton-iso/issues/153

### PR DESCRIPTION
**Pull request by mico**
closes #238

### CODE REVIEW

File: spec/fixtures/iso-pubid-directives.txt [hunk 1]
```diff
@@ -2,8 +2,11 @@ ISO/IEC DIR 1
 ISO/IEC DIR 1:2022
 ISO/IEC DIR 2
 ISO/IEC DIR 2:2021
+ISO/IEC DIR 2 IEC
+ISO/IEC DIR 2 ISO
```
Comment (andrew2net) [inline spec/fixtures/iso-pubid-directives.txt:6]: We also have year editions `ISO/IEC DIR 1 ISO:2002`. Add it to the test, please.

**Review by andrew2net — CHANGES_REQUESTED** [review]
It fails to parse `ISO/IEC DIR JTC 1 SUP` but the test doesn't detect the issue. Please fix the test.
```ruby
Pubid::Iso::Identifier.parse 'ISO/IEC DIR JTC 1 SUP'
NoMethodError: undefined method `map' for " SUP"@17:Parslet::Slice
from /Users/andrej/RubyProjects/pubid-iso/lib/pubid/iso/transformer.rb:19:in `block in <class:Transformer>'
```

File: spec/fixtures/iso-pubid-directives.txt (hunk 1 above)
Comment (mico) [reply spec/fixtures/iso-pubid-directives.txt:7]: @andrew2net done

**Review by andrew2net — APPROVED** [review]
LGTM

**Comment by andrew2net** [comment]
@mico please release this update. It blocks addressing the [issue](https://github.com/relaton/relaton-iso/issues/153).

**Pull request merged**